.github/workflows/release.yml

```diff
@@ -4,31 +4,15 @@ on:
   release:
     types:
       - published
+      - prereleased
 
 jobs:
-  pypi-publish:
-    name: Upload release to TestPyPI
-    runs-on: ubuntu-latest
-    environment:
-      name: pypi
+  release:
     permissions:
-      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
-      contents: read
-    steps:
-    # retrieve your distributions here
-      - uses: actions/checkout@v3
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.x'
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install build
-      - name: Build package
-        run: python -m build
-      - name: Publish package distributions to TestPyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
-          repository-url: https://test.pypi.org/legacy/
+      id-token: write
+      contents: write
+    uses: SonarSource/gh-action_release/.github/workflows/main.yaml@v5
```

Comment:
```suggestion
    uses: SonarSource/gh-action_release/.github/workflows/main.yaml@feat/jd/BUILD-4665-python-releasability
```

Reply:
I'll keep v5 here for future maintenance purposes. The working branch was tested with sonar-dummy-python-oss